libs/api/domains/regulations-admin/src/lib/graphql/regulationsAdmin.resolver.ts

```diff
@@ -0,0 +1,260 @@
+} from './dto'
+import {
+  DeleteDraftRegulationModel,
+  DraftRegulationCancelModel,
+  DraftRegulationChangeModel,
+  DraftRegulationPdfDownloadModel,
+  DraftRegulationShippedModel,
+} from './models'
+import { CreateDraftRegulationInput } from './dto/createDraftRegulation.input'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver()
+export class RegulationsAdminResolver {
+  constructor(
+    private regulationsService: RegulationsService,
+    private regulationsAdminApiService: RegulationsAdminApi,
+    private regulationsAdminClientService: RegulationsAdminClientService,
+    @Inject(DownloadServiceConfig.KEY)
+    private downloadServiceConfig: ConfigType<typeof DownloadServiceConfig>,
+  ) {}
+
+  // @Query(() => DraftRegulationModel, { nullable: true })
+  @Query(() => graphqlTypeJson)
+  async getDraftRegulation(
+    @Args('input') input: GetDraftRegulationInput,
+    @CurrentUser() user: User,
+  ) {
+    return await this.regulationsAdminClientService.getDraftRegulation(
+      input.draftId,
+      user.authorization,
+    )
+  }
+
+  @Query(() => graphqlTypeJson)
+  async getRegulationImpactsByName(
+    @Args('input') input: GetRegulationImpactsInput,
+    @CurrentUser() user: User,
+  ) {
+    return ensureRegName(input.regulation)
+      ? await this.regulationsAdminApiService.getImpactsByName(
+          input.regulation,
+          user.authorization,
+        )
+      : null
+  }
+
+  @Query(() => [DraftRegulationShippedModel])
+  async getShippedRegulations(@CurrentUser() { authorization }: User) {
+    return await this.regulationsAdminClientService.getShippedRegulations(
+      authorization,
+    )
+  }
+
+  // @Query(() => [DraftRegulationSummaryModel])
+  @Query(() => graphqlTypeJson)
+  async getDraftRegulations(
+    @Args('input') input: GetDraftRegulationsInput,
+    @CurrentUser() user: User,
+  ) {
+    return await this.regulationsAdminClientService.getDraftRegulations(
+      user.authorization,
+      input.page,
+    )
+  }
+
+  @Mutation(() => graphqlTypeJson)
+  async createDraftRegulation(
+    @Args('input') input: CreateDraftRegulationInput,
+    @CurrentUser() { authorization }: User,
+  ) {
+    return this.regulationsAdminApiService.create(authorization, input)
+  }
+
+  @Mutation(() => graphqlTypeJson)
+  async updateDraftRegulationById(
+    @Args('input') input: EditDraftRegulationInput,
+    @CurrentUser() { authorization }: User,
+  ) {
+    return this.regulationsAdminApiService.updateById(
+      input.id,
+      input.body,
+      authorization,
+    )
+  }
+
+  @Mutation(() => DeleteDraftRegulationModel)
+  async deleteDraftRegulation(
+    @Args('input') input: DeleteDraftRegulationInput,
+    @CurrentUser() { authorization }: User,
+  ): Promise<DeleteDraftRegulationModel> {
+    await this.regulationsAdminApiService.deleteById(
+      input.draftId,
+      authorization,
+    )
+
+    return {
+      id: input.draftId,
+    }
+  }
+
+  @Query(() => graphqlTypeJson)
```

Comment:
We should try to type all of these.
This will require a bit of refactoring, I think we should keep it for now and I will update it in a following PR, just so we can get this to main and start the testing phase with the project owners.

I have added a task to our board.